src/main/java/org/zalando/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -73,14 +73,29 @@ public Subscription createSubscription(final SubscriptionBase subscriptionBase)
 
             return subscription;
         } catch (final JsonProcessingException e) {
-            throw new InconsistentStateException("Serialization problem during persistence of event type", e);
+            throw new InconsistentStateException("Serialization problem during persistence of subscription", e);
         } catch (final DuplicateKeyException e) {
             throw new DuplicatedSubscriptionException("Subscription with the same key properties already exists", e);
         } catch (final DataAccessException e) {
             throw new RepositoryProblemException("Error occurred when running database request", e);
         }
     }
 
+    public void updateSubscription(final Subscription subscription) {
+        final String keyFieldsHash = hashGenerator.generateSubscriptionKeyFieldsHash(subscription);
```

Comment:
@antban I think we need to only allow changes to `authorization` section. Because if some user changes `owning_application`, for example, then the result of `keyFieldsHash` might not even match any existing row or worse, match a different subscription.

Reply:
@antban forget about it. I saw down there that you check it all.